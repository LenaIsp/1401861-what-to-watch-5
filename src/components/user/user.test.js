import React from "react";
import renderer from "react-test-renderer";
import User from "./user";
import {TEST_MOCK_STORE, TEST_MOCK_USER} from "../../test-mock.js";
import {MemoryRouter} from "react-router-dom";
import configureMockStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureMockStore();
const store = mockStore(TEST_MOCK_STORE);

describe(`User`, () => {
  it(`Should App render correctly`, () => {
    const tree = renderer
    .create(
        <Provider store={store}>
          <MemoryRouter>
            <User
              userStatus={TEST_MOCK_USER.userStatus}
              avatarUrl={TEST_MOCK_USER.avatarUrl}
            />,
          </MemoryRouter>
        </Provider>
    )
  .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
