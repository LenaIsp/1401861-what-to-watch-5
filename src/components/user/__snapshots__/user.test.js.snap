// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Should App render correctly 1`] = `
Array [
  <div
    className="user-block"
  >
    <a
      href="/mylist"
      onClick={[Function]}
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg"
          width="63"
        />
      </div>
    </a>
  </div>,
  ",",
]
`;
